src/server/scripts/Shadowlands/SepulcherOfTheFirstOnes/boss_anduin_wrynn.cpp

```diff
@@ -0,0 +1,4307 @@
+    {
+        _scheduler.Update(diff);
+    }
+
+    void HandleWickedStarHit(Unit* unit)
+    {
+        if (unit->GetEntry() == NPC_SYLVANAS_WINDRUNNER_ANDUIN || unit->GetEntry() == NPC_UTHER_THE_LIGHTBRINGER_ANDUIN ||
+            unit->GetEntry() == NPC_LADY_JAINA_PROUDMOORE_ANDUIN || unit->GetEntry() == BOSS_ANDUIN_WRYNN)
+            return;
+
+        if (Unit* caster = at->GetCaster())
+        {
+            if (std::find(_affectedUnits.begin(), _affectedUnits.end(), unit->GetGUID()) == _affectedUnits.end())
+            {
+                if (caster->IsValidAttackTarget(unit))
+                    caster->CastSpell(unit, SPELL_WICKED_STAR_DAMAGE_SILENCE, CastSpellExtraArgs(TriggerCastFlags(TRIGGERED_IGNORE_GCD | TRIGGERED_IGNORE_CAST_IN_PROGRESS)));
+                else if (caster->IsValidAssistTarget(unit))
+                    caster->CastSpell(unit, SPELL_WICKED_STAR_EMPOWERMENT, CastSpellExtraArgs(TriggerCastFlags(TRIGGERED_IGNORE_GCD | TRIGGERED_IGNORE_CAST_IN_PROGRESS)));
+
+                _affectedUnits.push_back(unit->GetGUID());
+            }
+        }
+    }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        HandleWickedStarHit(unit);
+    }
+
+    void OnUnitExit(Unit* unit) override
+    {
+        HandleWickedStarHit(unit);
+    }
+
+    virtual void HandleNextMovement()
+    {
+        ReturnToCaster();
+    }
+
+    void OnDestinationReached() override
+    {
+        Unit* caster = at->GetCaster();
+        if (!caster)
+            return;
+
+        if (at->GetDistance(_casterInitialPosition) > 0.05f)
+        {
+            _affectedUnits.clear();
+            HandleNextMovement();
+        }
+        else
+            at->Remove();
+    }
+
+    void ReturnToCaster()
+    {
+        uint32 starReturnSpeed = 10;
+
+        switch (at->GetMap()->GetDifficultyID())
+        {
+            case DIFFICULTY_LFR_NEW:
+                starReturnSpeed = 8;
+                break;
+            case DIFFICULTY_NORMAL_RAID:
+                starReturnSpeed = 10;
+                break;
+            case DIFFICULTY_HEROIC_RAID:
+                starReturnSpeed = 12;
+                break;
+            case DIFFICULTY_MYTHIC_RAID:
+                starReturnSpeed = 14;
+                break;
+            default:
+                starReturnSpeed = 10;
+                break;
+        }
+
+        _scheduler.Schedule(0ms, [this, starReturnSpeed](TaskContext task)
+        {
+            Movement::PointsArray returnSplinePoints;
+
+            returnSplinePoints.push_back(PositionToVector3(at));
+            returnSplinePoints.push_back(PositionToVector3(at));
+            returnSplinePoints.push_back(PositionToVector3(_casterInitialPosition));
+            returnSplinePoints.push_back(PositionToVector3(_casterInitialPosition));
+
+            at->InitSplines(returnSplinePoints, static_cast<uint32>(at->GetDistance(_casterInitialPosition) / starReturnSpeed * 600));
+            task.Repeat(250ms);
```

Comment:
since I know you copied this script from my Divine Star script, my question is whether it is true that this areatrigger only has 4 points repeating every 250ms when returning.

Reply:
should have 4 points for empowered only, regular wicked stars should have 5 - nice catch!